### ISSUE TITLE
ERROR: 0:5: '' : unary '-' not allowed for provided data type

### ISSUE THREAD

**Issue by paulthomson** [mc-imperial/shader-compiler-bugs#4]
[[Shaders, screenshots, error log, etc.]](https://github.com/mc-imperial/shader-compiler-bugs/tree/master/Intel-HD-Graphics-520-Windows/compile_errors/large-v440-1f1265b42fca6aab_inv_variant_12) [[ZIP]](https://minhaskamal.github.io/DownGit/#/home?url=https://github.com/mc-imperial/shader-compiler-bugs/tree/master/Intel-HD-Graphics-520-Windows/compile_errors/large-v440-1f1265b42fca6aab_inv_variant_12)

variant_12_reduced_338_simplified.frag:

```glsl
#version 440

void f()
{
    - (vec2(0.0, 0.0) * mat4x2(0.0));
}
```

Compile error log:

```
ERROR: 0:5: '' : unary '-' not allowed for provided data type 
ERROR: 0:5: '' : unary '-' not allowed for provided data type 
```

**Comment by paulthomson** [comment]
Reported:
https://software.intel.com/en-us/forums/graphics-driver-bug-reporting/topic/703553

**Comment by afd** [comment]
Fixed by Intel in 21.20.16.4542.
